### PR TITLE
Task/unit test project

### PR DESCRIPTION
**Pull request by CampbellJGilpin**

### CODE REVIEW

File: v2/AdminCore.WebApi.Tests/Controllers/ClientControllerTests.cs [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+      _clientService.GetAll().Returns(clients);
+
+      _mapper.Map<IList<ClientDto>, List<ClientViewModel>>(Arg.Is(clients)).Returns(clientViewModels);
+
+      // Act
+      var result = _controller.GetAllClients();
+
+      // Assert
+      Assert.True(result is List<ClientDto>);
```
Comment (BrownMark) [inline v2/AdminCore.WebApi.Tests/Controllers/ClientControllerTests.cs]: You need to use the new method on the base controller.

File: v2/AdminCore.WebApi.Tests/Controllers/DashboardControllerTest.cs [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+
+      _dashboardService.GetTeamSnapshotDashboardEvents().Returns(snapshotReturnedFromService);
+
+      _mapper.Map<Dictionary<string, List<EmployeeSnapshotDto>>, List<TeamSnapshotViewModel>>(snapshotReturnedFromService).Returns(snapshotModel);
+
+      // Act
+      var result = _dashboardController.GetDashboardSnapshot();
+
+      // Assert
+      RetrieveValueFromResult<List<TeamSnapshotViewModel>>(result);
```
Comment (BrownMark) [inline v2/AdminCore.WebApi.Tests/Controllers/DashboardControllerTest.cs]: Verify data is returned. var x = RetrieveValueFromResult<List<TeamSnapshotViewModel>>(result);

File: v2/AdminCore.WebApi.Tests/Controllers/DashboardControllerTest.cs [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+
+      _mapper.Map<IList<EventDto>, List<DashboardEventViewModel>>(Arg.Is(eventsReturnedFromService)).Returns(eventModels);
+
+      _employeeCredentials.GetUserId().Returns(employeeId);
+
+      // Act
+      var result = _dashboardController.GetEmployeeEvents(searchDate);
+
+      // Assert
+      RetrieveValueFromResult<EmployeeEventViewModel>(result);
```
Comment (BrownMark) [inline v2/AdminCore.WebApi.Tests/Controllers/DashboardControllerTest.cs]: Again as above

File: v2/AdminCore.WebApi.Tests/Controllers/DashboardControllerTest.cs [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+
+      _mapper.Map<Dictionary<string, List<EmployeeSnapshotDto>>, List<TeamSnapshotViewModel>>(snapshotReturnedFromService).Returns(snapshotModels);      
+
+      _employeeCredentials.GetUserId().Returns(employeeId);
+
+      // Act
+      var result = _dashboardController.GetEmployeeTeamSnapshot();
+
+      // Assert
+      RetrieveValueFromResult<List<EmployeeEventViewModel>>(result);
```
Comment (BrownMark) [inline v2/AdminCore.WebApi.Tests/Controllers/DashboardControllerTest.cs]: I'll comment this on, but needs done throughout.

File: v2/AdminCore.WebApi.Tests/Controllers/TeamControllerTests.cs [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+      // Arrange
+      var teamToSaveModel = _fixture.Build<CreateTeamViewModel>()
+        .With(x => x.teamName, string.Empty)
+        .Create();
+
+      // Act
+      var result = _teamController.CreateTeam(teamToSaveModel);
+
+      // Assert
+      AssertObjectResultIsNull<TeamDto>(result);
```
Comment (BrownMark) [inline v2/AdminCore.WebApi.Tests/Controllers/TeamControllerTests.cs]: Base method

**Review by BrownMark — CHANGES_REQUESTED** [review]

File: v2/AdminCore.WebApi.Tests/Controllers/ClientControllerTests.cs (hunk 1 above)
Comment (CampbellJGilpin) [reply v2/AdminCore.WebApi.Tests/Controllers/ClientControllerTests.cs]: Updated

File: v2/AdminCore.WebApi.Tests/Controllers/TeamControllerTests.cs (hunk 5 above)
Comment (CampbellJGilpin) [reply v2/AdminCore.WebApi.Tests/Controllers/TeamControllerTests.cs]: This is from the BaseControllerTest class

File: v2/AdminCore.WebApi.Tests/Controllers/DashboardControllerTest.cs (hunk 2 above)
Comment (CampbellJGilpin) [reply v2/AdminCore.WebApi.Tests/Controllers/DashboardControllerTest.cs]: Updated - Setting what's returned and when applicable checking the count is correct.

File: v2/AdminCore.WebApi.Tests/Controllers/DashboardControllerTest.cs (hunk 3 above)
Comment (CampbellJGilpin) [reply v2/AdminCore.WebApi.Tests/Controllers/DashboardControllerTest.cs]: Updated - Setting what's returned and when applicable checking the count is correct.

File: v2/AdminCore.WebApi.Tests/Controllers/DashboardControllerTest.cs (hunk 4 above)
Comment (CampbellJGilpin) [reply v2/AdminCore.WebApi.Tests/Controllers/DashboardControllerTest.cs]: Updated - Setting what's returned and when applicable checking the count is correct.

**Comment by CampbellJGilpin** [comment]
Updated several places to fix return type from controllers being set as DTO

**Comment by CampbellJGilpin** [comment]
Merged conflicts and made updates based on review. Should be ready for review again @BrownMark @eoinmcafee

**Comment by CampbellJGilpin** [comment]
Ready for review again. Resolved merge and made updates from previous review. @BrownMark @eoinmcafee

**Review by BrownMark — APPROVED** [review]

**Pull request merged**